### PR TITLE
Polygon Intersection Bug Fix

### LINKED ISSUE #14477: Polygon Intersection Bug
We observed this problem via ConvertToReflectometryQ.
This was reported by Jos. The following script:

```
r=s.Runs()
dref=s.OurLoad(r[36265,36266,36267,36268])
MoveInstrumentComponent(dref,'DetectorBench',Y=0.0404)
#dtrans=s.OurLoad(r[36252])

Transo=CreateTransmissionWorkspaceAuto(dtrans,AnalysisMode="MultiDetectorAnalysis",ProcessingInstructions="380-428",WavelengthMin=2.0)
IvsQ2o, IvsLam2o, thetaOut=ReflectometryReductionOneAuto(dref,AnalysisMode="MultiDetectorAnalysis",ProcessingInstructions="390-450", FirstTransmissionRun=Transo, ThetaIn="0.64",WavelengthMin=2.0,CorrectionAlgorithm='None',StrictSpectrumChecking=False,CorrectDetectorPositions=False)#,NormalizeByIntegratedMonitors=False
IvsQo, IvsLamo, thetaOut=ReflectometryReductionOneAuto(dref,AnalysisMode="MultiDetectorAnalysis",ProcessingInstructions=','.join(map(str, range(250, 558))), FirstTransmissionRun=Transo, ThetaIn="0.64",WavelengthMin=2.0,CorrectionAlgorithm='None',StrictSpectrumChecking=False,CorrectDetectorPositions=False)#,NormalizeByIntegratedMonitors=False
TvLo = ConvertSpectrumAxis(IvsLamo,'SignedTheta')

ConvertToReflectometryQ(TvLo,IncidentTheta=0.64,DumpVertexes=True,OverrideIncidentTheta=True, OutputWorkspace="QxQz",OutputVertexes="Qtableo",Extents=[-1.2e-6,-6.67e-7,0.01034,0.010388],NumberBinsQz=4,NumberBinsQx=4,Method='NormalisedPolygon',OutputAsMDWorkspace=False)
```
produces a NaN value (white bin):

bug

The bug is caused by the `V2D` equality operator, which uses a standard tolerance value `1e-6` (too high in some cases). This makes `PolygonIntersection` fail when calculating the intersection if one of the polygons is small enough.

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
Fixes #14477 

For tester: The script in the original issue should no longer produce a NaN.

Release notes: http://www.mantidproject.org/index.php?title=Release_Notes_3_6_Reflectometry&diff=25601&oldid=25589

### CODE REVIEW

File: Framework/Kernel/test/V2DTest.h [hunk 1]
```diff
@@ -197,6 +197,14 @@ class V2DTest : public CxxTest::TestSuite {
     // Symmetric in arguments
     TS_ASSERT_DELTA(second.angle(first), M_PI / 3.0, DBL_EPSILON);
   }
+
+  void test_Equality_Operator() {
```
Comment (OwenArnold) [inline Framework/Kernel/test/V2DTest.h]: Good

**Comment by OwenArnold** [comment]
This is exactly as I would expect to see the fix. Although I'm going to change the issue and PR description because this change goes beyond the ConvertToReflectometryQ.

**Comment by raquelalvarezbanos** [comment]
BASISAutoReductionTest is failing but the output and reference workspaces are almost identical. As an example, below is the comparison for spectrum 6.

comparison

**Comment by OwenArnold** [comment]
@raquelalvarezbanos The fix is great. I've tested it out and it's working fine. The difference to the BASIS auto reduction plot look too small to be real. I think you should go ahead update the reference results. Inform Pete about the changes, but I see no reason why a delta this small should cause problems.

**Comment by raquelalvarezbanos** [comment]
@OwenArnold I've just informed Pete and Andrei about the changes (and I'll inform Jose too)

**Comment by OwenArnold** [comment]
Many thanks for fixing this.

**Pull request merged**